table/server/common/src/main/java/alluxio/table/common/udb/UdbMountSpec.java

```diff
@@ -0,0 +1,589 @@
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nullable;
+
+/**
+ * Tables and partitions inclusion and exclusion specification.
+ */
+public final class UdbMountSpec {
+  /**
+   * Tables and partitions to bypass.
+   */
+  private final TablePartitionWrapper mBypassed;
+
+  /**
+   * Ignored tables.
+   * Tables are ignored as a whole, no partition configuration is needed.
+   */
+  private final SimpleWrapper mIgnored;
+
+  protected UdbMountSpec(TablePartitionWrapper bypassed,
+                         SimpleWrapper ignored) {
+    mBypassed = bypassed;
+    mIgnored = ignored;
+  }
+
+  /**
+   * Checks if a table should be bypassed.
+   *
+   * @param tableName the table name
+   * @return true if the table is configured to be bypassed, false otherwise
+   * @see UdbMountSpec#hasFullyBypassedTable(String)
+   */
+  public boolean hasBypassedTable(String tableName) {
+    return !hasIgnoredTable(tableName) && mBypassed.hasTable(tableName);
+  }
+
+  /**
+   * Checks if all partitions of a table should be bypassed.
+   *
+   * @param tableName the table name
+   * @return true if the table is configured to be fully bypassed, false otherwise
+   * @see UdbMountSpec#hasBypassedTable(String)
+   */
+  public boolean hasFullyBypassedTable(String tableName) {
+    return !hasIgnoredTable(tableName) && mBypassed.hasFullTable(tableName);
+  }
+
+  /**
+   * Checks if a table is ignored.
+   *
+   * @param tableName the table name
+   * @return true if the table is ignored, false otherwise
+   */
+  public boolean hasIgnoredTable(String tableName) {
+    return mIgnored.has(tableName);
+  }
+
+  /**
+   * Checks by a partition's name if it should be bypassed.
+   *
+   * @param tableName the table name
+   * @param partitionName the partition name
+   * @return true if the partition should be bypassed, false otherwise
+   */
+  public boolean hasBypassedPartition(String tableName, String partitionName) {
+    return !hasIgnoredTable(tableName) && mBypassed.hasPartition(tableName, partitionName);
+  }
+
+  /**
+   * Wrapper class that takes into account inclusion and exclusion behaviour.
+   */
+  abstract static class InclusionExclusionWrapper<T extends NamePatternWrapper> {
+    protected final T mIncluded;
+    // excluded entries don't allow partition specifications -- they are name and regex only
+    protected final SimpleNamePatternWrapper mExcluded;
+
+    InclusionExclusionWrapper(T included, SimpleNamePatternWrapper excluded) {
+      // currently either included or excluded list is accepted, but not both
+      Preconditions.checkArgument(included.isEmpty() || excluded.isEmpty());
+      mIncluded = included;
+      mExcluded = excluded;
+    }
+
+    // generic implementation
+    boolean has(String name) {
+      if (mExcluded.isEmpty()) {
+        return mIncluded.has(name);
+      } else {
+        return !mExcluded.has(name);
+      }
+    }
```

Comment:
Since it cannot be an inclusion + exclusion list at the same time, the simplified inclusion/exclusion logic is simple: in case of an inclusion list, check if the name is present; in case of an exclusion list, check if the name is absent.